src/components/ImageView/index.js

```diff
@@ -46,16 +46,20 @@ class ImageView extends PureComponent {
     }
 
     componentDidMount() {
-        Image.getSize(this.props.url, (width, height) => {
-            this.setImageRegion(width, height);
-        });
         if (this.canUseTouchScreen) {
             return;
         }
         document.addEventListener('mousemove', this.trackMovement);
         document.addEventListener('mouseup', this.trackPointerPosition);
     }
 
+    componentDidUpdate() {
+        // Resize the image when cycled through
+        Image.getSize(this.props.url, (width, height) => {
+            this.setImageRegion(width, height);
+        });
```

Comment:
It seems that images get resized multiple times if we cycle fast. Maybe debouncing this method would fix it?

https://user-images.githubusercontent.com/22219519/210862947-060aa118-4f09-42a9-8775-ed2a8ab3766e.mov